File: AlgorithmicToolbox/week2/Assignment2/lcm/java/Lcm.java
// package LCM;
import java.util.*;

public class Lcm {
    public static long gcd(long a,long b)
    {
        if (b == 0)
            return a;
        return gcd(b, a % b);
    }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        long a = sc.nextLong();
        long b = sc.nextLong();
        long hcf = gcd(a, b);
        // System.out.println(a*b);
        // System.out.println(hcf);
        // System.out.println(((Object)(((a*b)/hcf))).getClass().getSimpleName());
        System.out.println(((a * b) / hcf));
    }
}

File: AlgorithmicToolbox/week2/Assignment2/lcm/cpp/lcm.cpp
#include <iostream>

using namespace std;

long long gcd(long long a, long long b)
{
    if (b == 0)
        return a;
    return gcd(b, a % b);
}

int main()
{
    long long a;
    long long b;
    cin>>a;
    cin >> b;
    long long hcf = gcd(a, b);
    cout << (a * b) / hcf ;
    return 0;
}